resources/validations/bin/validate_with_schematron.sh

```diff
@@ -1,69 +1,126 @@
 #!/usr/bin/env bash
-
 set -o pipefail
 
-if [ ! -e  "$1" ]; then
+# output root defaults to report folder relative to this script
+OUTPUT_ROOT="report/schematron"
+# schematron directory validate the file with each .sch found defaults to src/*.sch relative to this script
+SCHEMA_LOCATION_DIR="src"
+##
+## options ###################################################################
+##
+while echo "$1" | grep -- ^- > /dev/null 2>&1; do
+    case "$1" in
+        # input file to validate
+        -f)
+            shift
+            DOC_TO_VALIDATE="$1"
+            ;;
+        # saxon version
+        -v)
+            if test -n "$SAXON_CP"; then
+                echo "SAXON_CP is set to ${SAXON_CP} as an environment variable setting version using -v is invalid"
+                exit 1
+            else
+                shift
+                SAXON_VERSION="$1"
+            fi
+            ;;
+        # schema directory location
+        -s)
+            shift
+            SCHEMA_LOCATION_DIR="$1"
+            ;;
+        # output directory root
+        -o)
+            shift
+            OUTPUT_ROOT="$1"
+            ;;
+        # Unknown option!
+        -*)
+            usage "Error: Unknown option: $1"
+            exit 1
+            ;;
```

Comment:
Can we add a HEREDOC with help content argument, `-h`? I can help you set that up, but will need to read up on approach as you do not use `getopt` which I am more familiar with from copy-pasting. :-)

Reply:
i think we can do that, the runXspec.sh has good examples of echoing usage. I heavily cribbed the arg processing from thar